### PR TITLE
Add metadata and adjust precision according to locale 

### PR DESCRIPTION
**Pull request by immerrr**
This PR adds the following:
- read budget & file metadata
- adjust precision according to budget locale

### CODE REVIEW

**Comment by immerrr** [comment]
Initially I thought about taking those currencies straight from YNAB, but they might not be license-clean.  There are better sources of that info, both in terms of license conditions and completeness, e.g. http://cldr.unicode.org/index/downloads, but they don't necessarily match YNAB data.

File: ynab/schema.py [hunk 1]
```diff
@@ -24,10 +24,27 @@ class Entity(Model):
...
+    strictBudget = StringType()
+
+
+class FileMetaData(Model):
+    budgetDataVersion = StringType()
+    currentKnowledge = StringType()
+    entityType = StringType()
+
+
+class DeleteableEntity(Entity):
```
Comment (aldanor) [inline ynab/schema.py]: `DeletableEntity` maybe?

File: ynab/ynab.py [hunk 2]
```diff
@@ -108,12 +109,17 @@ def from_json(cls, data):
...
+        # TODO:
+        # - scheduledTransactions
+        # - accountMappings
+        # - monthlyBudgets
         self._accounts = Accounts._from_flat(self, data['accounts'])
         self._payees = Payees._from_flat(self, data['payees'])
         self._master_categories = MasterCategories._from_flat(self, data['masterCategories'])
         self._transactions = Transactions._from_flat(self, data['transactions'])
         self._transactions.sort_by('date')
+        self._meta_data = BudgetMetaData._from_flat(self, data['budgetMetaData'])
```
Comment (aldanor) [inline ynab/ynab.py:121]: Maybe just `_metadata`? Double underscores looks quite ugly

**Comment by aldanor** [comment]
Thanks, this looks reasonable. I would be inclined to actually add the currencies though before merging this in. And they have to match YNAB's locale, so the only way is probably to pull it out of YNAB.

I don't think there should be any license problems, it's not really proprietary information to start with...

File: ynab/ynab.py (hunk 2 above)
Comment (immerrr) [inline ynab/ynab.py:121]: There's a precedent with `_master_categories` above, so not sure about that, I was just following the capitalization of the field from the JSON. But if you insist, I'll rename it.

File: ynab/schema.py (hunk 1 above)
Comment (immerrr) [inline ynab/schema.py]: yup

File: ynab/ynab.py (hunk 2 above)
Comment (aldanor) [inline ynab/ynab.py:121]: Ah, that's fine then, nevermind.

**Comment by immerrr** [comment]
Bumped.

**Comment by aldanor** [comment]
Looks good, thanks!

**Pull request merged**